live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/properties/Repository.java

```diff
@@ -25,7 +25,17 @@
  */
 package org.alfresco.hxi_connector.live_ingester.adapters.config.properties;
 
+import java.util.Objects;
 import jakarta.validation.constraints.NotBlank;
 
-public record Repository(@NotBlank String endpoint, @NotBlank String discoveryEndpoint)
-{}
+public record Repository(@NotBlank String endpoint, String discoveryEndpoint, String versionOverride)
+{
+    public Repository
+    {
+        if ((Objects.isNull(discoveryEndpoint) || discoveryEndpoint.isBlank()) &&
+                (Objects.isNull(versionOverride) || versionOverride.isBlank()))
+        {
+            throw new IllegalStateException("Discovery endpoint or version override must be set in the Live Ingester configuration.");
```

Comment:
Maybe refer to the actual properties instead of their logical name? E.g.: `alfresco.repository.discovery-endpoint / alfresco.repository.version-override`

This would make it easier for an admin to fix immediately

Reply:
Done.